apollo-client/src/test/java/com/ctrip/framework/apollo/spring/JavaConfigAnnotationTest.java

```diff
@@ -91,6 +108,74 @@ public void testApolloConfigWithInheritance() throws Exception {
     assertEquals(applicationConfig, bean.getSomeConfig());
   }
 
+  @Test
+  public void testEnableApolloConfigResolveExpressionSimple() {
+    String someKey = "someKey-2020-11-14-1750";
+    String someValue = UUID.randomUUID().toString();
+    mockConfig(ConfigConsts.NAMESPACE_APPLICATION, mock(Config.class));
+    Config xxxConfig = mock(Config.class);
+    when(xxxConfig.getProperty(eq(someKey), anyString())).thenReturn(someValue);
+    mockConfig("xxx", xxxConfig);
+
+    TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration configuration =
+        getSimpleBean(TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration.class);
+
+    // check
+    assertEquals(someValue, configuration.getSomeKey());
+    verify(xxxConfig, times(1)).getProperty(eq(someKey), anyString());
+  }
+
+  @Test
+  public void testEnableApolloConfigResolveExpressionFromSystemProperty() {
+    mockConfig(ConfigConsts.NAMESPACE_APPLICATION, mock(Config.class));
+    final String someKey = "someKey-2020-11-14-1750";
+    final String someValue = UUID.randomUUID().toString();
+
+    final String propertyKey = "simple.namespace";
+    final String resolvedNamespaceName = "yyy";
+    System.setProperty(propertyKey, resolvedNamespaceName);
+
+    Config yyyConfig = mock(Config.class);
+    when(yyyConfig.getProperty(eq(someKey), anyString())).thenReturn(someValue);
+    mockConfig(resolvedNamespaceName, yyyConfig);
+
+    TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration configuration =
+        getSimpleBean(TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration.class);
+
+    // check
+    assertEquals(someValue, configuration.getSomeKey());
+    verify(yyyConfig, times(1)).getProperty(eq(someKey), anyString());
+
+    System.clearProperty(propertyKey);
+  }
+
+  @Test(expected = BeanCreationException.class)
+  public void testEnableApolloConfigUnresolvedValueInField() {
+    mockConfig(ConfigConsts.NAMESPACE_APPLICATION, mock(Config.class));
+    mockConfig("xxx", mock(Config.class));
+    getSimpleBean(TestEnableApolloConfigResolveExpressionWithDefaultValueConfiguration.class);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testEnableApolloConfigUnresolvable() {
+    getSimpleBean(TestEnableApolloConfigUnresolvableConfiguration.class);
+  }
+
+  /**
+   * Could not resolve placeholder.
+   * Unfortunately, when use {@link EnableApolloConfig},
+   * only key in {@link System#getenv(String)} or {@link System#getProperty(String)} can be resolved.
+   */
+  @Test(expected = IllegalArgumentException.class)
+  public void testEnableApolloConfigResolveFromNamespaceApplication() {
```

Comment:
I think this test is not necessary...
Besides, the description is not accurate because there are many places to configure the property, e.g. application.properties

Reply:
Deleted it.